crsf.md

```diff
@@ -371,6 +373,24 @@ Such constants give ±2500cm/s range and 3cm/s precision at low speeds and 70cm/
     int16_t origin_address;             // Origin Device address
 ```
 
+## 0x0C RPM
+
+Frame type used to transmit RPM (revolutions per minute) telemetry data from the craft to the transmitter. This frame can be used to report motor or propeller RPM for monitoring performance or diagnostics.
+
+```cpp
+    uint8_t    rpm_source_id;  // Identifies the source of the RPM data (e.g., 0 = Motor 1, 1 = Motor 2, etc.)
+    int24_t    rpm_value;      // RPM with positive and negative values (e.g., motor running in forward or reverse)
+```
+
+## 0x0D TEMP
+
+Frame type used to transmit temperature telemetry data from the vehicle to the transmitter. This frame can be used to report temperature readings from various sources on the vehicle, such as motors, ESCs, or the environment.
+
+```cpp
+    uint8_t     temp_source_id;     // Identifies the source of the temperature data (e.g., 0 = Motor 1, 1 = ESC, 2 = Ambient, etc.)
```

Comment:
```suggestion
    uint8_t     temp_source_id;     // Identifies the source of the temperature data (e.g., 0 = FC including all ESCs, 1 = Ambient, etc.)
```

Reply:
FC **including all ESCs**? The rest of the suggested changes make perfect sense (packing the values to optimise bandwidth), but how does this work? Each id should map to a single measurement, not sum of multiple (since that becomes a nonsense  value).

Reply:
the temperature values has also been modified to be a an array. so the idea would be that the FC fills said array with _singular values_ not a sum.

Reply:
Usually by directly attaching said device to a receiver. Both ELRS and Crossfire can talk directly to VTXes now. AM32 is working on a direct attach solution as well. 
VBat sensing on the RX has been a feature for > 10years.

Reply:
My planes don't have VTXes and I was talking about a multi-cell battery measurement and possible multiple ESCs (multi motor plane)

Reply:
ok, i think i understand where you're coming from, the idea is _not_ that the receiver starts collecting entries from different devices and regenerate a singular table. 

in a multi-device scenario, each device should still have its own id assigned and if they only have one temperature sensor also just include one value, the idea is really just to allow devices that possess multiple sensors compress those into a single packet.

eg.

```
id 1 - temperature sensor fuselage
id 2 - temperature sensor lipo
id 3 - turbine
       - sensor 1 temperature inlet
       - sensor 2 temperature exhaust
```

as for addressing tx-side, yeah some algorithm needs to be found there.
i was kinda thinking `id * 100 + array index`. so turbine exhaust would be id 301.

Reply:
as the ids are might getting used by more sensors, maybe add another telegram like "id proposal" with sub telegrams id request ( id) and response from other devices, maybe only when this address is taken